src/expression.d

```diff
@@ -14633,11 +14633,34 @@ extern (C++) final class PowExp : BinExp
         if (checkArithmeticBin())
             return new ErrorExp();
 
+        return rewritePowHack(this, sc);
+    }
+
+    override void accept(Visitor v)
+    {
+        v.visit(this);
+    }
+}
+
+/**
+ This function tries to simplify and rewrite ^^-expressions 
+ a ^^ (2 || 3) turns into (a*a) or (a*a*a) respectivly 
+ a ^^ b turns into std.math.pow(a,b)
+ (This is a rather nasty hack and should be replace by using builtins
+  or something similar)
+*/
+private static Expression rewritePowHack(PowExp powexp, Scope* sc)
```

Comment:
This should be called something else, such as `rewritePowerAsMults`, it is not a hack, but it could be made more useful.

For instance, Knuth's optimal power tree in "Seminumerical algorithms".